Dining/models.py

```diff
@@ -48,11 +48,11 @@ class DiningList(models.Model):
     adjustable_duration = models.DurationField(
         default=settings.TRANSACTION_PENDING_DURATION,
         help_text="The amount of time the dining list can be adjusted after its date")
-    claimed_by = models.ForeignKey(settings.AUTH_USER_MODEL, blank=True, related_name="dininglist_claimer", null=True,
-                                   on_delete=models.SET_NULL)
+    claimed_by = models.ForeignKey(settings.AUTH_USER_MODEL, related_name="dininglist_claimer",
```

Comment:
Also, I prefer this being on_delete=models.Set_Null opposed to protect. For counting it doesn't matter who claims a dining list, but on behalf of what association. I don't want dining lists to be accidentally deleted in an attempt to delete a user.

Reply:
What about #130? Set null is not possible if null is not allowed.

Reply:
130 doesn't state anywhere why claimed_by can NOT by null, so why change it to a non-nullable object?

Reply:
Oh that relates more to #129. In short: it makes it more complex if you each time when using `claimed_by` you need to think about that it can be `None`. Usually having fields required instead of optional makes it much easier to reason about. The `get_purchaser` method can be much easier now that we don't have to check for that and it is definitely very advantageous that it can't return different types of objects anymore (it would've returned an association previously instead of a user when claimed_by was `None`, I'm not sure if the application can handle that).

Next to this, I don't see that I'll ever accept a change that makes it possible for associations to own a dining list. It makes it way too complex. I'd very much rather simply allow multiple owners as in #130. We can simply write a function that make all board members of an association owner of a certain dining list, to mimic the same behavior that you want.